File: connecthub-backend/schemas/match.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from .user import User

class MatchBase(BaseModel):
    user_id: int
    matched_user_id: int

class MatchCreate(MatchBase):
    pass

class LastMessage(BaseModel):
    text: str
    sent_at: datetime
    is_from_current_user: bool

class Match(MatchBase):
    id: str
    user: User
    matched_at: datetime
    last_message: Optional[LastMessage] = None
    is_new_match: bool = True

    class Config:
        from_attributes = True

File: connecthub-frontend/src/types/match.ts
import type { AppUser } from './user'

export interface Match {
  id: string
  user_id: number
  matched_user_id: number
  user: AppUser
  matched_at: string
  last_message?: {
    text: string
    sent_at: string
    is_from_current_user: boolean
  }
  is_new_match: boolean
}

export interface MatchStatus {
  matches: Match[]
  totalCount: number
}